src/core/ComposableElement.hack

```diff
@@ -294,28 +294,27 @@ abstract xhp class x:composable_element extends :xhp {
    *   <foo attr1="bar" {...$xhp} />
    *
    * This will only copy defined attributes on $xhp to when they are also
-   * defined on $this. "Special" data-/aria- attributes will still need to be
-   * implicitly transferred, since the typechecker never knows about them.
+   * defined on $this, or if they are "special" data-/aria- attributes.
    *
    * Defaults from $xhp are copied as well, if they are present.
```

Comment:
Does the change from `__xhpReflectionAttributes()` to `getAttributes()` affect this? Either way can you add an attribute with a default value to the test so that we test whatever the correct behavior is?

Reply:
nice catch, fixed on update

Reply:
```
    $x = <test:default_attributes />;
    $y = <test:attribute_types mystring="foo" {...$x} />;
    expect($y->:mystring)->toEqual('mydefault');
```

I'll follow up with @ngavalas on this one

Reply:
Checked with njg, this is correct: both options (`mydefault`, `foo`) are sometimes reasonable, sometimes surprising, but `mydefault` is consistent.

This behavior would also be consistent if we migrate attributes to properties with default.